modules/services/espanso.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs, config, lib, ... }:
+let
+  inherit (lib) mkOption mkEnableOption mkIf maintainers;
+  inherit (pkgs) espanso xclip;
+  cfg = config.services.espanso;
+  yaml = pkgs.formats.yaml { };
+in {
+  meta.maintainers = with maintainers; [ lucasew ];
+
+  options = {
+    services.espanso = {
+      enable = mkEnableOption "Espanso: cross platform text expander in Rust";
+      config = mkOption {
+        type = yaml.type;
+        description = "Espanso configuration";
+        default = {
+          matches = [
+            { # Simple text replacement
+              trigger = ":espanso";
+              replace = "Hi there!";
+            }
+            { # Dates
+              trigger = ":date";
+              replace = "{{mydate}}";
+              vars = [{
+                name = "mydate";
+                type = "date";
+                params = { format = "%m/%d/%Y"; };
+              }];
+            }
+            { # Shell commands
+              trigger = ":shell";
+              replace = "{{output}}";
+              vars = [{
+                name = "output";
+                type = "shell";
+                params = { cmd = "echo Hello from your shell"; };
+              }];
+            }
+          ];
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
```

Comment:
Still it would be better to have a specific assertion in this case so it has a better error for users trying to use this module.

We can always remove later on once someone adds support to Darwin.

Reply:
I've done an assert that only passes if it's in linux